tests/test_enterprise/api/test_views.py

```diff
@@ -2696,6 +2700,226 @@ def test_bulk_enrollment(self, body, expected_code, expected_body):
+        """
+        factories.EnterpriseCustomerFactory(
+            uuid=FAKE_UUIDS[0],
+            name="test_enterprise"
+        )
+
+        permission = Permission.objects.get(name='Can add Enterprise Customer')
+        self.user.user_permissions.add(permission)
+
+        response = self.client.post(
+            settings.TEST_SERVER + ENTERPRISE_CUSTOMER_BULK_ENROLL_LEARNERS_IN_COURSES_ENDPOINT,
+            data=json.dumps(body),
+            content_type='application/json',
+        )
+        self.assertEqual(response.status_code, expected_code)
+        if expected_response:
+            response_json = response.json()
+            self.assertEqual(expected_response, response_json)
+        self.assertEqual(len(PendingEnrollment.objects.all()), expected_num_pending_licenses)
+
+    @mock.patch('enterprise.api.v1.views.EnterpriseCustomerViewSet._create_ecom_orders_for_enrollments')
+    @mock.patch('enterprise.api.v1.views.enroll_licensed_users_in_courses')
+    # pylint: disable=unused-argument
+    def test_enroll_learners_in_courses_partial_failure(self, mock_enroll_user, mock_ecom_order):
+        """
+        Tests that bulk users bulk enrollment endpoint properly handles partial failures.
+        """
+        ent_customer = factories.EnterpriseCustomerFactory(
+            uuid=FAKE_UUIDS[0],
+            name="test_enterprise"
+        )
+
+        permission = Permission.objects.get(name='Can add Enterprise Customer')
+        self.user.user_permissions.add(permission)
+
+        pending_ecu, __ = PendingEnterpriseCustomerUser.objects.get_or_create(
+            enterprise_customer=ent_customer,
+            user_email='abc@test.com'
+        )
+
+        course = 'course-v1:edX+DemoX+Demo_Course'
+        enrollment_response = {
+            'pending': [{'email': 'abc@test.com', 'course_run_key': course, 'user': pending_ecu}],
+            'successes': [],
+            'failures': [{'email': 'xyz@test.com', 'course_run_key': course}]
+        }
+        mock_enroll_user.return_value = enrollment_response
+
+        body = {
+            'licenses_info': [
+                {
+                    'email': 'abc@test.com',
+                    'course_run_key': course,
+                    'course_mode': 'verified',
+                    'license_uuid': '5a88bdcade7c4ecb838f8111b68e18ac'
+                },
+                {
+                    'email': 'xyz@test.com',
+                    'course_run_key': course,
+                    'course_mode': 'verified',
+                    'license_uuid': '2c58acdade7c4ede838f7111b42e18ac'
+                },
+            ]
+        }
+
+        response = self.client.post(
+            settings.TEST_SERVER + ENTERPRISE_CUSTOMER_BULK_ENROLL_LEARNERS_IN_COURSES_ENDPOINT,
+            data=json.dumps(body),
+            content_type='application/json',
+        )
+        self.assertEqual(response.status_code, 409)
```

Comment:
SUGGEST: Using the status enums from rest_framework e.g., https://github.com/edx/edx-enterprise/blob/b55a50473775be18505565b99d7966389fb81ca9/tests/test_consent/api/test_permissions.py#L47